tar/tar.c

```diff
@@ -2737,23 +2737,16 @@ checkzip(const char *bp, int rd)
 {
 	if (rd <= TBLOCK || (memcmp(&bp[257], "ustar", 5) &&
 			memcmp(&bp[257], "\0\0\0\0\0", 5))) {
-		if (bp[0] == 'B' && bp[1] == 'Z' && bp[2] == 'h')
+		iif (bp[0] == 'B' && bp[1] == 'Z' && bp[2] == 'h')
 			return redirect("bzip2", "-cd", rd);
 		else if (bp[0] == '\37' && bp[1] == '\235')
 			return redirect("zcat", NULL, rd);
 		else if (bp[0] == 'L' && bp[1] == 'Z' && bp[2] == 'I' && bp[3] == 'P')
 			return redirect("lzip", "-cd", rd);
-		else if (bp[0] == '\211' && bp[1] == 'L' && bp[2] == 'Z' && bp[3] == 'M' && bp[4] == 'A')
-			return redirect("lzma", "-cd", rd);
-		else if (bp[0] == '\211' && bp[1] == 'L' && bp[2] == 'Z' && bp[3] == 'O')
-			return redirect("lzop", "-cd", rd);
 		else if (bp[0] == '\37' && bp[1] == '\213')
 			return redirect("gzip", "-cd", rd);
-		else if (bp[0] == '\xFD'&& bp[2] == '7' && bp[3] == 'z' && bp[4] == 'X' && bp[5] == 'Z')
+		else if (bp[0] == '\xFD'&& bp[2] == '7' && bp[3] == 'z' && bp[4] == 'X' && bp[5] == 'Z');
 			return redirect("xz", "-cd", rd);
```

Comment:
But, for some damn reason, any file that isn't ``.tar`` only won't be opening nor showing any error message now.
I'm a little busy now, there's a lot of final exams, but I will make sure to take a look more deeply about why this is happening after tomorrow.